### PR TITLE
Add check for hands when unbuckling

### LINKED ISSUE #28097: Creatures without hands are able to unbuckle other creatures
## Description

Had a round on mander where I played as a mouse. I found a catatonic player hiding in the bathroom, and found that I was able to unbuckle them from the toilet even though I have no hands. As expected, I was unable to buckle them back into their seat, as I had no hands.

**Reproduction**

Attempt to unbuckle a buckled creature as a mouse.
**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by akiwa**
## About the PR

Only entities that have hands can now unbuckle other entities.

## Why / Balance

Closes #28097. It makes sense that you need hands to unbuckle someone (as the same logic already exists for buckling someone).

## Technical details

An additional check was added at the beginning of the TryUnbuckle method that checks if the entity of the person trying to unbuckle someone has a HandsComponent attached to it. If not a popup is given to the user (analogous to buckling) and the method returns false.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

CantUnbuckleWithoutHands

## Breaking changes

**Changelog**

:cl:Rieschla
- tweak: Entities without hands now cannot unbuckle other entities.

### CODE REVIEW

**Comment by Cojoke-dot** [comment]
This would block the space dragon from getting out of a chair if someone were to lock them in it.

How does this affect mob ai? Can they still get out of chairs whenever they don't have hands?

**Comment by Partmedia** [comment]
Thanks for the fix. Would one of the existing checks in `ActionBlockerSystem` be more suitable for this check?

File: Content.Shared/Buckle/SharedBuckleSystem.Buckle.cs [hunk 1]
```diff
@@ -484,6 +484,14 @@ private bool CanUnbuckle(Entity<BuckleComponent?> buckle, EntityUid? user, bool
...
             return false;
 
+        if (user.HasValue && !HasComp<HandsComponent>(user))
+        {
+            // PopupPredicted when
+            if (_netManager.IsServer)
+                _popup.PopupEntity(Loc.GetString("buckle-component-no-hands-message"), user.Value, user.Value);
+            return false;
+        }
+
```
Comment (metalgearsloth) [inline Content.Shared/Buckle/SharedBuckleSystem.Buckle.cs:494]: As notafet said actionblocker is far more appropriate, also looks like you just copy-pasted this, as I wrote this comment a few years ago, but predicted popups have existed for years.

**Comment by metalgearsloth** [comment]
Re-open if you come back to it.

**Pull request closed without merging**